js/src/redux/providers/status.js

```diff
@@ -27,20 +27,49 @@ export default class Status {
     this._longStatus = {};
     this._minerSettings = {};
 
-    this._longStatusTimeoutId = null;
+    this._timeoutIds = {};
+    this._blockNumberSubscriptionId = null;
 
     this._timestamp = Date.now();
+
+    api.transport.on('close', () => {
+      this.stop();
+
+      const apiStatus = this.getApiStatus();
+      this._apiStatus = apiStatus;
+      this._store.dispatch(statusCollection(apiStatus));
+    });
+
+    api.transport.on('open', () => {
+      this.start();
+    });
   }
 
   start () {
+    this.stop();
+
     this._subscribeBlockNumber();
-    this._pollStatus();
-    this._pollLongStatus();
+    this._pollLongStatus(true);
```

Comment:
In theory (before I run this), we should now detect the chain change basically immediately? (Instead of the long delay as it is in current master.)